x/incentives/keeper/distribute.go

```diff
@@ -516,18 +511,64 @@ func (k Keeper) syncVolumeSplitGroup(ctx sdk.Context, group types.Group) error {
 
 	// Update group's total weight
 	updatedGroup.InternalGaugeInfo.TotalWeight = totalWeight
+	return updatedGroup, nil
+}
+
+// syncVolumeSplitGroup syncs a group according to volume splitting policy.
+// It mutates the passed in object and sets the updated value in state.
+// If there is an error, the passed in object is not mutated.
+//
+// It returns an error if:
+// - the volume for any linked pool is zero or cannot be found
+// - the cumulative volume for any linked pool has decreased (should never happen)
+func (k Keeper) syncVolumeSplitGroup(ctx sdk.Context, group types.Group) error {
+	updatedGroup, err := k.updatedGroupWeights(ctx, group)
+	if err != nil {
+		return err
+	}
 
 	k.SetGroup(ctx, updatedGroup)
 
 	// We return zero here so that the Group with zero total weight is silently skipped in the
 	// caller distribution logic.
-	if totalWeight.IsZero() {
+	if updatedGroup.InternalGaugeInfo.TotalWeight.IsZero() {
 		return types.GroupTotalWeightZeroError{GroupID: group.GroupGaugeId}
 	}
 
 	return nil
 }
 
+// QueryVolumeSplitGroup calculates the ratio of volume for each gauge in a group since the last epoch.
+// It first updates the group weights based on the pool volumes.
+// Then, for each gauge in the updated group, it calculates the ratio of the gauge's current weight to the total weight of the group.
+// If the total weight of the group is zero, the ratio of volume for the gauge is set to zero.
+// The function returns a slice of GaugeVolume, each representing a gauge and its ratio of volume.
+// It returns an error if there is an issue updating the group weights.
+func (k Keeper) QueryVolumeSplitGroup(ctx sdk.Context, group types.Group) ([]types.GaugeVolume, error) {
```

Comment:
Let's keep query specifc logic in `grpc_query.go` and defined on the queried itself

Reply:
moved query logic to `grpc_query.go` here https://github.com/osmosis-labs/osmosis/pull/6651/commits/8285fad2f482eb931fcc1679d6d4ca0e06f69c7b